library/src/containers/AsyncApi/AsyncApi.tsx

```diff
@@ -16,7 +22,8 @@ import ErrorComponent from '../Error/Error';
 import { AsyncApiWrapper } from './styled';
 
 export interface AsyncApiProps {
-  schema: string | Object;
+  schema?: string | Object;
+  urlSchema?: FetchingSchemaInterface;
```

Comment:
Can we enforce that either schema or urlSchema must be provided? I know it's possible with prop-types, but not sure about Typescript.

Reply:
I didn't explain myself correctly. I mean having both schema and urlSchema, like it's right now, but enforcing in the code the rule of having just one of them.

Reply:
I implemented my proposition from second comment in this thread. Please see: https://github.com/asyncapi/asyncapi-react/pull/34/files#diff-2fa873daa631d9246738398633704fd9R25

Reply:
Nice! I love this solution. Just one question: I see it says `string | Object | FetchSchemaInterface`, will it always fallback into `Object`? Saying because an object of the form FetchSchemaInterface is also an object, right? Or will typescript take care of this?

Reply:
Thanks :) You can pass schema as string (json or yaml), or Object (this is reference to AsyncAPIInterface) or FetchSchemaInterface. I agree with you that FetchSchemaInterface is also a Object, but Object with appropriate fields, so if dev pass Object as FetchSchemaInterface to component then component will execute whole code as FetchSchemaInterface, if pass as Object, then code validate this Object as AsyncApi and show errors if this spec is invalid :) You have to look at it in terms of static typing, not dynamic.

Reply:
Understood. Thanks! :)